src/platform/Darwin/DnssdContexts.cpp

```diff
@@ -568,8 +560,9 @@ bool ResolveContext::TryReportingResultsForInterfaceIndex(uint32_t interfaceInde
         return false;
     }
 
-    auto & interface = interfaces[interfaceIndex];
-    auto & ips       = interface.addresses;
+    std::pair<uint32_t, std::string> interfaceKey = std::make_pair(interfaceIndex, this->domainName);
```

Comment:
I don't get this part.  Why is this using `this->domainName`?  Are all the things in the map for the same domain?

Reply:
This is incorrect. Initially I was getting the domain from the host name  :) but when i sync'ed with the tip, I saw the changes from you - https://github.com/project-chip/connectedhomeip/commit/000edb450b7c975ff0ab0f60060f0605b75d8a44, and did a hasty fix. :(